actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -363,7 +363,7 @@ def content_tag(name, content_or_options_with_block = nil, options = nil, escape
       #   token_list(nil, false, 123, "", "foo", { bar: true })
       #    # => "123 foo bar"
       def token_list(*args)
-        tokens = build_tag_values(*args).flat_map { |value| value.to_s.split(/\s+/) }.uniq
+        tokens = build_tag_values(*args).flat_map { |value| CGI.unescape_html(value.to_s).split(/\s+/) }.uniq
```

Comment:
Since reaching outside of Action View feels a little odd, I also experimented with a version of this change that mapped over the tokens with [escape_once](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-escape_once), then chained `.join(" ").html_safe` instead of passing the value to `safe_join`. 

That felt like it'd confuse the matter of HTML safety, and would feel more risky than prepping the values _before_ passing them to [safe_join](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/OutputSafetyHelper.html#method-i-safe_join).